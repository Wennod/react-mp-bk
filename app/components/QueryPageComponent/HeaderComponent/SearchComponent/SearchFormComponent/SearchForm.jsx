import React from 'react';

class SearchForm extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <form onSubmit={this.props.getMovies}>
                <label htmlFor="fieldId">Find your movies</label>
                <input 
                    onChange={() => this.props.setQueryValue(this.inputField)} 
                    ref={el => this.inputField = el}  
                    className="search-field" 
                    type="text" 
                    name="search" 
                    id="fieldId" 
                />
                <div className="search-filter">
                    <p className="search-filter__title">Search By</p>
                    <input 
                        onClick={() => this.props.setQueryParam(this.titleRadio)} 
                        ref={el => this.titleRadio = el} 
                        className="search-filter__toggler" 
                        type="radio" 
                        name="searchCriteria" 
                        id="title" 
                        value="title" 
                        defaultChecked 
                    />
                    <input 
                        onClick={() => this.props.setQueryParam(this.genreRadio)}
                        ref={el => this.genreRadio = el} 
                        className="search-filter__toggler" 
                        type="radio" 
                        name="searchCriteria" 
                        id="genre" 
                        value="genres" 
                    />
                </div>
                <input className="search-button" type="submit" value="SEARCH"/>
            </form>
        )
    }
}

export default SearchForm;