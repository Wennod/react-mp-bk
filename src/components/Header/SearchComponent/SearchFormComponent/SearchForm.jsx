import React from 'react';

import Style from './SearchForm.style.scss';

class SearchForm extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <form className="search-form" onSubmit={this.props.getMovies}>
                <h2>Find your movies</h2>
                <input 
                    onChange={() => this.props.setQueryValue(this.inputField)} 
                    ref={el => this.inputField = el}  
                    className="search-form__input" 
                    type="text" 
                    name="search" 
                    id="fieldId" 
                />
                <div className="search-form__group">
                    <div className="search-form__filter">
                        <p className="search-form__filter-title">Search By</p>
                        <input 
                            onClick={() => this.props.setQueryParam(this.titleRadio)} 
                            ref={el => this.titleRadio = el} 
                            className="search-form__filter-toggler search-form__filter-toggler--t" 
                            type="radio" 
                            name="searchCriteria" 
                            id="title" 
                            value="title" 
                            defaultChecked 
                        />
                        <input 
                            onClick={() => this.props.setQueryParam(this.genreRadio)}
                            ref={el => this.genreRadio = el} 
                            className="search-form__filter-toggler search-form__filter-toggler-g" 
                            type="radio" 
                            name="searchCriteria" 
                            id="genre" 
                            value="genres" 
                        />
                    </div>
                    <input className="search-button" type="submit" value="SEARCH"/>
                </div>
            </form>
        )
    }
}

export default SearchForm;