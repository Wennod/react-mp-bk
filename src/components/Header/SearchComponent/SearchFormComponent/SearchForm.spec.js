import React from 'react';
import SearchForm from './SearchForm';

describe('SearchForm snapshot', () => {
    it('should match snapshot', () => {
        const tree = mount(<SearchForm />);
        expect(tree).toMatchSnapshot();
    });
});