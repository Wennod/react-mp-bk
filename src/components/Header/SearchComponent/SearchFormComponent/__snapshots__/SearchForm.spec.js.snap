// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm snapshot should match snapshot 1`] = `
<SearchForm>
  <form
    className="search-form"
  >
    <h2>
      Find your movies
    </h2>
    <input
      className="search-form__input"
      id="fieldId"
      name="search"
      onChange={[Function]}
      type="text"
    />
    <div
      className="search-form__group"
    >
      <div
        className="search-form__filter"
      >
        <p
          className="search-form__filter-title"
        >
          Search By
        </p>
        <input
          className="search-form__filter-toggler search-form__filter-toggler--t"
          defaultChecked={true}
          id="title"
          name="searchCriteria"
          onClick={[Function]}
          type="radio"
          value="title"
        />
        <input
          className="search-form__filter-toggler search-form__filter-toggler-g"
          id="genre"
          name="searchCriteria"
          onClick={[Function]}
          type="radio"
          value="genres"
        />
      </div>
      <input
        className="search-button"
        type="submit"
        value="SEARCH"
      />
    </div>
  </form>
</SearchForm>
`;
